client/packages/invoices/src/Prescriptions/LineEditView/AllocationSection.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import {
+  useTranslation,
+  TableProvider,
+  createTableStore,
+  createQueryParamsStore,
+  Grid,
+  useAuthContext,
+} from '@openmsupply-client/common';
+import { AutoAllocateField, AllocateInSelector } from '../../StockOut';
+import { AccordionPanelSection } from './PanelSection';
+import { PrescriptionLineEditTable } from './PrescriptionLineEditTable';
+import { AutoAllocatePrescribedQuantityField } from './AutoAllocatePrescribedQuantityField';
+
+export const AllocationSection = ({ disabled }: { disabled?: boolean }) => {
+  const t = useTranslation();
+  const { store: { preferences } = {} } = useAuthContext();
+
+  const showPrescribedQuantity =
+    preferences?.editPrescribedQuantityOnPrescription ?? true;
+
+  return (
+    <>
+      {showPrescribedQuantity && <AutoAllocatePrescribedQuantityField />}
+      <Grid display="flex" alignItems="center" gap={1}>
+        <AutoAllocateField
+          inputColor="white"
+          allowPartialPacks
+          autoFocus={!showPrescribedQuantity}
+        />
+        <AllocateInSelector />
+      </Grid>
+      <AccordionPanelSection
+        title={t('label.batches')}
+        defaultExpanded={false}
+        // key={key + '_table'} // Do we need to use a custom key again? item_id?
```

Comment:
@CarlosNZ is this important, I assume it was added for a reason

Reply:
Oh lol sorry, nah think we're all good here, parent component already has key on it, to be re-rendered when item ID changes 🙏